examples/cookies.js

```diff
@@ -103,23 +103,22 @@ export default async function () {
       },
     ]);
     cookies = await context.cookies("http://foo.com", "https://baz.com");
-    check(cookies.length, {
+    await check(cookies.length, {
       'number of filtered cookies should be 2': n => n === 2,
     });
-    check(cookies[0], {
+    await check(cookies[0], {
       'the first filtered cookie name should be foo': c => c.name === 'foo',
       'the first filtered cookie value should be 42': c => c.value === '42',
     });
-    check(cookies[1], {
+    await check(cookies[1], {
       'the second filtered cookie name should be baz': c => c.name === 'baz',
       'the second filtered cookie value should be 44': c => c.value === '44',
     });
 
     // clear cookies
     await context.clearCookies();
-    cookies = await context.cookies();
-    check(cookies.length, {
-      'number of cookies should be zero': n => n === 0,
+    await check(await context.cookies(), {
+      'number of cookies should be zero': c => c.length === 0,
```

Comment:
Seeing this, I realized that we could make this a bit nicer by allow the check value to be a promise. 

```ts
await check(context.cookies(), {
  // Branches can be synchronous, because they are executed after the value has resolved.
  'number of cookies should be zero': c => c.length === 0,
})
```

I have a [PR](https://github.com/grafana/k6-jslib-utils/pull/14) up.

Reply:
This is awesome! Thanks :) We can update the examples when the new lib version is released.